pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -38,6 +38,10 @@ const (
 	CatalogSourceRegistryServerError ConditionReason = "RegistryServerError"
 )
 
+type IndexImageSecret struct {
+	// SecretName is the name of the kubernetes Secret object.
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
IMO this field would be a bit repetitive on the spec, with something like 
```
secretName:
  imageSecret: <my-secret-name>
```
would something more generic like `auth` on the spec be more extensible? 
```
auth:
   catalogPullSecret: <my-secret-name>
```

Reply:
@exdx I was trying to come up with something better than those two too. 

FYI right now it's not 

```
secretName: 
    imageSecret: <my-secret-name>
```

It's 
```
imageSecret: 
     secretName: <my-secret-name>
```
But something like 
```
auth:
     imageSecret: <my-secret-name>
```
maybe more intuitive? 
`catalogPullSecret` may sound like it's asking the user to plug in the registry account pull secret there instead of mentioning the secret name.